moveit_ros/hybrid_planning/local_planner/local_planner_component/src/local_planner_component.cpp

```diff
@@ -235,15 +241,12 @@ void LocalPlannerComponent::executeIteration()
     // If the planner received an action request and a global solution it starts to plan locally
     case LocalPlannerState::LOCAL_PLANNING_ACTIVE:
     {
-      // Read current planning scene
+      // Read current robot state
       planning_scene_monitor_->updateSceneWithCurrentState();
       planning_scene_monitor_->lockSceneRead();  // LOCK planning scene
-      planning_scene::PlanningScenePtr planning_scene = planning_scene_monitor_->getPlanningScene();
+      const auto current_robot_state = planning_scene_monitor_->getPlanningScene()->getCurrentState();
       planning_scene_monitor_->unlockSceneRead();  // UNLOCK planning scene
 
-      // Get current state
-      auto current_robot_state = planning_scene->getCurrentStateNonConst();
-
```

Comment:
This was a clear bug in the previous code and the fix looks good :+1: 
Still `LockedPlanningSceneRO` should be strongly preferred over the `lock`/`unlock` methods...

Reply:
Thanks. Maybe `lockSceneRead() / lockSceneWrite()` should be private, then?  (in a follow-up PR, of course)

Reply:
YES. Though you will have to add friends to the locker to get there.
Locking the PSM can most likely be refactored quite a bit more with modern coding practice. the whole `LockedPlanningSceneR*` structure seems rather messy..